src/devices/cpu/ht1130/ht1130.h

```diff
@@ -0,0 +1,127 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+
+#ifndef MAME_CPU_HT1130_HT1130_H
+#define MAME_CPU_HT1130_HT1130_H
+
+#pragma once
+
+
+class ht1130_device : public cpu_device
+{
+public:
+	ht1130_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+
+	enum {
+		HT1130_PC,
+		HT1130_R0,
+		HT1130_R1,
+		HT1130_R2,
+		HT1130_R3,
+		HT1130_R4,
+		HT1130_ACC,
+		HT1130_TIMER_EN,
+		HT1130_TIMER,
+	};
+
+	auto pm_in_cb() { return m_port_in_pm.bind(); }
+	auto ps_in_cb() { return m_port_in_ps.bind(); }
+	auto pp_in_cb() { return m_port_in_pp.bind(); }
+
+	auto pa_out_cb() { return m_port_out_pa.bind(); }
+
+	auto display_offset_out_cb() { return m_display_offset_out.bind(); }
+	auto display_data_out_cb() { return m_display_data_out.bind(); }
```

Comment:
Is this the best way of exposing this?  The chip stores the state of the 4*32 LCD segments internally, and provides a fixed duty cycle output to drive an LCD panel directly.  It doesn’t expose the internal RAM layout.

But even if you want to expose display RAM writes, you can use a single `devcb_write8` and call it with the offset and data.  The device callbacks all support offset and mask, even if people don’t use them a lot of the time.

Reply:
It might not be ideal, but I've simplified it.

Best reworked by somebody with a better understanding how how these usually get hooked up (I was following another driver when I did this)